### DISCUSSION TITLE
Can't process the remote service request

### DISCUSSION THREAD

**Discussion by robertosanti** [redisson/redisson#5049] [Q&A]
Hi All
we have an application deployed on 4 containers in AWS.  Every node has 2  different instances of Redisson connected to a Redis cluster.  We use Redisson for locks, blocking queues, RPC, and Spring Cache Manager.  The Redis cluster is composed of two slaves and the master.  As mentioned before the Redis cluster is entirely managed by AWS.
We have a problem with Remote Service. 
In a quartz job, only one node of our application's cluster registers the service. This is the snippet of code:

        RLock redisLock = this.redissonClient.getLock(LOCK_NAME);

        logger.info("Trying to acquire Redis lock...");

        if (!redisLock.tryLock()) {
            return;
        }

        logger.info("Redis lock acquired!");
        RRemoteService remoteService = redissonClient.getRemoteService();
        remoteService.register(ServiceClass.class, this.serviceInstance, 10);

The "redissonClient" instance and the "serviceInstance" are managed by Spring (XML configuration).
When we need to invoke this service we use this code:

        RRemoteService remoteService = redissonClient.getRemoteService();
        ServiceClass service = remoteService.get(ServiceClass.class);

The problem appears suddenly and not in all nodes:

2023-05-18 11:58:31,497 268493249 ERROR [org.redisson.RedissonRemoteService] (redisson-timer-7-1:) {} Can't process the remote service request with id 724cd8168825d870f962ac338394714f
org.redisson.client.RedisTimeoutException: Unable to acquire connection! java.util.concurrent.CompletableFuture@64a5fd85[Completed exceptionally: java.util.concurrent.CancellationException]Increase connection pool size. Node source: NodeSource [slot=0, addr=null, redisClient=null, redirect=null, entry=null], command: (EVAL), params: [local v = redis.call('hget', KEYS[1], ARGV[1]); redis.call('hdel', KEYS[1], ARGV[1]); return v, 1, {redisson_rs:com.fake.ServiceClass }:tasks, PooledUnsafeDirectByteBuf(ridx: 0, widx: 32, cap: 256)] after 3 retry attempts
	at org.redisson.command.RedisExecutor$1.run(RedisExecutor.java:258)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.run(HashedWheelTimer.java:715)
	at io.netty.util.concurrent.ImmediateExecutor.execute(ImmediateExecutor.java:34)
	at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:703)
	at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:790)
	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:503)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)

Could someone help us figure out where we're going wrong?
Many thanks
Redisson version 3.20.0
Redis version: 6.2.6

**Comment by mrniko** [comment]
You need to shutdown RRemoteService if it's not used anymore since it's allocates a connection for serving requests. Which may cause connection pool starvation

**Comment by robertosanti** [comment]
Thanks for the reply!
How can I do this? I didn't find any methods in the RRemoteService interface to close or shutdown.
Is it good practice to use RRemoteService instance as a singleton object?
Thanks in the advance

**Reply by mrniko** [reply]
Sorry, you need to call `deregister` method.

**Comment by robertosanti** [comment]
Thank you very much for your reply
